### PR TITLE
Cloudsdk 75

### PR DESCRIPTION
**Pull request by szll**

### CODE REVIEW

File: lib/ExtensionConfigWatcher.js [hunk 1]
```diff
@@ -15,12 +15,13 @@ class ExtensionConfigWatcher extends EventEmitter {
...
-      fsEx.readJson(path, (err, config) => {
+    this.watcher = chokidar.watch(path.join(this.extensionFolder, '**', 'extension-config.json'), this.options).on('all', (event, configPath) => {
+      fsEx.readJson(configPath, (err, config) => {
         if (err) throw err
 
-        const parts = path.split('/')
-        const extPath = parts.slice(0, parts.length - 1).join('/')
+        const parts = configPath.split(path.sep)
+        const extPath = parts.slice(0, parts.length - 1).join(path.sep)
+        console.log(configPath, parts, path.sep)
```
Comment (daddy-cool) [inline lib/ExtensionConfigWatcher.js]: remove console.log

File: lib/ExtensionWatcher.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ class ExtensionWatcher extends EventEmitter {
     this.configPath = path.join(settingsFolder, EXTENSIONS_FILE)
     this.attachedExtensions = []
     this.options = options || {}
+    // this.options.ignoreInitial = true
```
Comment (daddy-cool) [inline lib/ExtensionWatcher.js]: remove line

File: lib/actions/BackendAction.js [hunk 3]
```diff
@@ -97,8 +97,8 @@ class BackendAction {
       times: 5,
       interval: (count) => 100 * Math.pow(2, count)
     }, (acb) => {
-      this.dcClient.updatePipeline(pipeline, AppSettings.getInstance().getId(), (err) => {
-        if (err) return acb(new Error(`Could not update pipeline '${pipeline.pipeline.id}'`))
+      this.dcClient.uploadPipeline(pipeline, AppSettings.getInstance().getId(), (err) => {
+        if (err) return acb(new Error(`Could not upload pipeline '${pipeline.pipeline.id}': ${err.message}`))
```
Comment (daddy-cool) [inline lib/actions/BackendAction.js]: log error (as debug)

File: test/actions/BackendAction-test.js [hunk 4]
```diff
@@ -57,19 +53,36 @@ describe('BackendAction', () => {
     backendAction.extensionConfigWatcher = {
       stop: (cb) => { cb() }
     }
+
+    process.env.APP_PATH = appPath
+    const appSettings = new AppSettings()
+    mkdirp.sync(path.join(appPath, AppSettings.SETTINGS_FOLDER))
+    appSettings.setId('foobarTest').setAttachedExtensions({}).save()
+
+    // TODO: thats a workaround for .save().init()
```
Comment (daddy-cool) [inline test/actions/BackendAction-test.js]: remove todo, add this line to already existing ticket in backlog

File: lib/app/backend/extensionRuntime/StepExecutor.js [hunk 5]
```diff
@@ -1,26 +1,33 @@
...
+      ],
+      ignoreInitial: true
+    }
   }
 
   start (cb) {
     if (this.childProcess) return cb(new Error('childProcess already running'))
 
-    this.childProcess = fork(program, [], options)
+    this.childProcess = fork(program, [], childProcessOptions)
```
Comment (daddy-cool) [inline lib/app/backend/extensionRuntime/StepExecutor.js]: set childProcess.stop=false here

File: lib/app/backend/extensionRuntime/StepExecutor.js [hunk 6]
```diff
@@ -54,6 +61,15 @@ class StepExecutor {
     this.watcher.on('ready', () => cb())
   }
 
+  stopWatcher (cb) {
+    if (!this.watcher) return cb()
```
Comment (daddy-cool) [inline lib/app/backend/extensionRuntime/StepExecutor.js]: add this.watcher.removeAllListeners

Comment (szll) [reply lib/app/backend/extensionRuntime/StepExecutor.js]: done

File: lib/app/backend/extensionRuntime/StepExecutor.js (hunk 5 above)
Comment (szll) [reply lib/app/backend/extensionRuntime/StepExecutor.js]: removed childProcess.stop=true instead

**Pull request merged**